cmake/config/debug.cmake

```diff
@@ -0,0 +1,29 @@
+SET(FLAGS
+    -Wextra
+    -Wno-unused-parameter
+    -Wno-deprecated-declarations
```

Comment:
Not really debug-specific, moreover you add it here:
https://github.com/hyperledger/iroha/blob/c1416b8135999ce6da45c6f746a06c29f1fbd963/cmake/config/all.cmake#L13